geoips/plugin_registry.py

```diff
@@ -144,59 +181,633 @@ def _set_class_properties(self):
             #         self._registered_plugins,
             #         "all_registered_plugins",
             #     )
-        return self._registered_plugins
 
-    def get_plugin_info(self, interface, plugin_name):
-        """Find a plugin in the registry and return its info.
+    @staticmethod
+    def _find_registry_files(namespace):
+        """Locate all plugin registry files found under 'namespace'.
+
+        Parameters
+        ----------
+        namespace: str
+            - The namespace in which plugin packages are registered to. Usually, this
+              will be the 'geoips.plugin_package' namespace, however this can be changed
+              by providing a different top-level namespace to the PluginRegistry class.
+
+        Returns
+        -------
+        registry_files: list[str]
+            - A list of filepaths corresponding to expected registered_plugins.json
+              files for all plugin packages found under namespace.
+
+        Raises
+        ------
+        PluginRegistryError:
+            - Occurs if a package is potentially missing an __init__.py file.
+        """
+        registry_files = []  # Collect the paths to the registry files here
+        for pkg in metadata.entry_points(group=namespace):
+            try:
+                registry_files.append(
+                    str(resources.files(pkg.value) / "registered_plugins.json")
+                )
+            except TypeError:
+                raise PluginRegistryError(
+                    f"resources.files('{pkg.value}') failed\n"
+                    f"pkg {pkg}\n"
+                    "Potentially missing __init__.py file? Try:\n"
+                    f"    touch {pkg.value}/__init__.py\n"
+                    "and try again.\n"
+                    "Note you will need to add a docstring to "
+                    f"{pkg.value}/__init__.py in order for all tests to pass"
```

Comment:
This is unrelated to this PR, but are these __init__.py files required in eery folder related to the requirement for "importlib" in the pytest.ini?  If we just enforce having an __init__.py file in every folder, will it remove the need for ipmortlib in the pytest.ini?

Reply:
For the record, I think it is fine to require __init__.py in every foldder